### ISSUE TITLE
Nothing to compile. Try running `make clean` first.

### ISSUE THREAD

**Issue by Ali-Lee-Kai-Sum** [facebook/infer#1167]
Hi guys, I meet a problem when I use infer in my own project, like this:

Nothing to compile. Try running `make clean` first.
There was nothing to analyze.

  No issues found

if I do "make clean", the same.

I found my object file is ARM aarch64, but my enevionment is x86, so I don't know whether it is the reason or not ?

Please hellp me, thanks in advance.

**server.c.o: ELF 64-bit LSB relocatable, ARM aarch64, version 1 (SYSV), not stripped**

**Comment by DGuco** [comment]
the same thing happened to me

**Comment by TomMD** [comment]
Infer keys in on a small set of possible compiler commands.  Cross compilers and versioned compilers (such as gcc-6.4.1) are not in its list and won't register as a compilation task to intercept and analyze.

**Comment by 251** [comment]
As I noticed today it's not just versions. `clang` works but a full path `/path/bin/clang` fails (v1.0.0).

**Comment by TomMD** [comment]
@251 Yes, that's a valuable observation.  In https://console.muse.dev we generate our own compilation database, look hard at the command to discern if it is a compilation command (handling versions, paths, etc) then dispatch to Infer and others.  You might benefit from doing something similar.  On the cheap you could use bear/compiledb and a basic script (at least a language with JSON support) to post process.

**Comment by 251** [comment]
@TomMD Thanks for the suggestion. In my case a modified `$PATH` sufficed as a quick fix.

**Comment by puyang0521** [comment]
> @TomMD Thanks for the suggestion. In my case a modified `$PATH` sufficed as a quick fix.

How to modify $PATH?

**Comment by TomMD** [comment]
@puyang0521 Try:

```
PATH=$PATH:/the/other/directory/you/want infer run -- make
```

**Comment by ichingking** [comment]
you need arm toolchain to compile infer.
